independent-projects/arc/runtime/src/main/java/io/quarkus/arc/NoInterceptors.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * If a business method is annotated {@code @NoInterceptors}, then class-level interceptors
+ * will be ignored for this method.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface NoInterceptors {
```

Comment:
This name is short and concise. That's good. OTOH we should probably also mention "class-level" and "bindings" and keep in mind that this annotation will not be used very often. So a longer name would not hurt. Maybe something like `@SkipClassBindings` or `@NoClassLevelBindings`?

Reply:
If you ask me, `@ExcludeClassInterceptors` is a perfect name :-) Though reusing that would probably be a bad idea. I'd be fine with `@NoClassInterceptors`. I agree that `@NoClassInterceptorBindings` would be too long, but if I were to choose between retaining the word "interceptor" and the word "binding", I'd always choose "interceptor", because everyone knows what that means. The fact that the annotation used to associate the interceptor with the target is called "interceptor binding" is an insignificant detail that's only important for spec lawyers :-)

Reply:
`@DisableClassInterceptors`?

Reply:
> but if I were to choose between retaining the word "interceptor" and the word "binding", I'd always choose "interceptor", because everyone knows what that means.

I tend to agree although I am aware that it is factually inaccurate :-/

Reply:
@mkouba because that way it is understandable and we can still say it is not our fault the name is inaccurate, right? :-D

But jokes aside, I don't think we should reuse it. The annotation has javadoc which specifically describes how it works for `@Interceptors`, see [this](https://github.com/eclipse-ee4j/interceptor-api/blob/master/api/src/main/java/jakarta/interceptor/ExcludeClassInterceptors.java#L30-L33). That IMO makes it all the more confusing for whoever happens to know the interceptors spec or read javadocs.

If anything, we should create new annotation with similar name - `@Skip/Disable/IgnoreClassAnnotations` and go from there.

Reply:
Well, the javadoc also mentions that it's non-portable for bindings and given the fact that it's a simple annotation with no members and our own javadoc is one line... But `@Skip/Disable/IgnoreClassAnnotations`  works for me.

Reply:
I don't have a strong opinion, so if you both agree on reusing existing annotation then by all means do that.
I just thought that having a dedicated annotation would be a 'cleaner' solution.

Reply:
Renamed to `@NoClassInterceptors`. I know it's disappointing, but you know I'm picky :-)